keyboards/ergodox_ez/keymaps/bpruitt-goddard/keymap.c

```diff
@@ -0,0 +1,234 @@
+// used to start recording a macro (double tap or more), to stop recording (any
+// number of tap), or to play the recorded macro (1 tap).
+void macro_tapdance_fn(qk_tap_dance_state_t *state, void *user_data) {
+  uint16_t keycode;
+  keyrecord_t record;
+  dprintf("macro_tap_dance_fn %d\n", state->count);
+  if (is_macro1_recording) {
+    keycode = DYN_REC_STOP;
+    is_macro1_recording = false;
+    layer_state_set_user(current_layer_state);
+  } else if (state->count == 1) {
+    keycode = DYN_MACRO_PLAY1;
+  } else {
+    keycode = DYN_REC_START1;
+    is_macro1_recording = true;
+    layer_state_set_user(current_layer_state);
+  }
+
+  record.event.pressed = true;
+  process_record_dynamic_macro(keycode, &record);
+  record.event.pressed = false;
+  process_record_dynamic_macro(keycode, &record);
+}
+
+// The definition of the tap dance actions:
+qk_tap_dance_action_t tap_dance_actions[] = {
+  // This Tap dance plays the macro 1 on TAP and records it on double tap.
+  [TAP_MACRO] = ACTION_TAP_DANCE_FN(macro_tapdance_fn)
+};
+
+// Runs for each key down or up event.
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  if (keycode != TD(TAP_MACRO)) {
+    // That key is processed by the macro_tapdance_fn. Not ignoring it here is
+    // mostly a no-op except that it is recorded in the macros (and uses space).
+    // We can't just return false when the key is a tap dance, because
+    // process_record_user, is called before the tap dance processing (and
+    // returning false would eat the tap dance).
+    if (!process_record_dynamic_macro(keycode, record)) {
+      return false;
+    }
+
+    if(record->event.pressed) {
+      switch(keycode) {
+        case MC_ARROW:
+          SEND_STRING("=>");
+          return false;
+          break;
+      }
+    }
+  }
+
+  return true; // Let QMK send the enter press/release events
+}
+
+// Runs just one time when the keyboard initializes.
+void matrix_init_user(void) {
+  ergodox_right_led_1_off();
+  ergodox_right_led_2_off();
+  ergodox_right_led_3_off();
+};
+
+// Runs constantly in the background, in a loop.
+void matrix_scan_user(void) {
+
+};
+
+// Value to use to switch LEDs on. The default value of 255 is far too bright.
+static const uint8_t max_led_value = 20;
+
+// Whether the given layer (one of the constant defined at the top) is active.
+#define LAYER_ON(layer) (current_layer_state & (1<<layer))
+
+void led_1_on(void) {
+  ergodox_right_led_1_on();
+  ergodox_right_led_1_set(max_led_value);
+}
+
+void led_2_on(void) {
+  ergodox_right_led_2_on();
+  ergodox_right_led_2_set(max_led_value);
+}
+
+void led_3_on(void) {
+  ergodox_right_led_3_on();
+  ergodox_right_led_3_set(max_led_value);
+}
+
+void led_1_off(void) {
+  ergodox_right_led_1_off();
+}
+
+void led_2_off(void) {
+  ergodox_right_led_2_off();
+}
+
+void led_3_off(void) {
+  ergodox_right_led_3_off();
+}
+
+uint32_t layer_state_set_user(uint32_t state) {
+  current_layer_state = state;
```

Comment:
This isn't actually needed.  You can use `layer_state`: https://github.com/qmk/qmk_firmware/blob/7c9d5ace143d3cc6d767a354acde814926d566fd/tmk_core/common/action_layer.c#L119

Reply:
But this is minor, and fine.